src/mscorlib/src/System/Threading/Tasks/ValueTask.CoreCLR.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Threading.Tasks
+{
+    readonly partial struct ValueTask<TResult>
+    {
+        private static void ThrowNullTaskException() => ThrowHelper.ThrowArgumentNullException(ExceptionArgument.task);
```

Comment:
It would be nice to have add ThrowHelper to the CoreFX package instead so that a copy of this method is not stamped into each ValueTask instantiaion.

Reply:
Is there an issue for that, or anyone working on it?

Reply:
I do not think so.

Reply:
Ok, I'll open an issue.  In the meantime I can just use ifdefs if you prefer.

Reply:
I do not understand why you would not want to just do this as part this change ...

Reply:
What about corert?

Reply:
Move to internal non-generic static class in meantime?

Reply:
Oh, were you suggesting adding a ThrowHelper to corefx rather than moving coreclr's implementation to shared?  I thought you were suggesting moving it to shared.  If you're just talking about a simple equivalent in corefx, sure, that's easy.

Reply:
Right, I am just talking about a simple local equivalent for ValueTuple in corefx. I do not think sharing ThrowHelper.cs makes sense.

Reply:
> I do not think sharing ThrowHelper.cs makes sense.

Right, that's why I was punting ;)